### DISCUSSION TITLE
How do I get the VolatileMemoryStore in the kernel?

### DISCUSSION THREAD

**Discussion by roalexan** [microsoft/semantic-kernel#1371] [Q&A]
Hello. I have a created a kernel, e.g.:

`            var kernel = new KernelBuilder().
                Configure(c =>
                {...})
            .WithMemoryStorage(**new VolatileMemoryStore()**)
            .Build();`

but later, I see that Kernel.Memory is an instance of: Microsoft.SemanticKernel.Memory.SemanticTextMemory.

How do I get back a reference to the VolatileMemoryStore in the kernel? I need it, because want to set a MemoryRecord into it via: UpsertAsync.

**Comment by craigomatic** [comment]
You would typically do this indirectly via the TextMemorySkill, ie: https://github.com/microsoft/semantic-kernel/blob/main/samples/dotnet/kernel-syntax-examples/Example15_MemorySkill.cs
